app.tsx

```diff
@@ -1,121 +1,161 @@
- 
-   const backgroundStyle = {
-     backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
-   };
- 
-   return (
-     <SafeAreaView style={backgroundStyle}>
-       <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
-       <ScrollView
-         contentInsetAdjustmentBehavior="automatic"
-         style={backgroundStyle}>
-         <View
-           style={{
-             backgroundColor: isDarkMode ? Colors.black : Colors.white,
-           }}>
-           <Section title="Bem-Vindo(a) à Taqtile!"/>
+import {Colors} from 'react-native/Libraries/NewAppScreen';
 
-           <Text style={styles.inputTitle}>E-mail</Text>
-           <TextInput style={styles.inputContainer}></TextInput>
+const Section: React.FC<
+  PropsWithChildren<{
+    title: string;
+  }>
+> = ({children, title}) => {
+  const isDarkMode = useColorScheme() === 'dark';
+  return (
+    <View style={styles.sectionContainer}>
+      <Text
+        style={[
+          styles.sectionTitle,
+          {
+            color: isDarkMode ? Colors.white : Colors.black,
+          },
+        ]}>
+        {title}
+      </Text>
+      <Text
+        style={[
+          styles.sectionDescription,
+          {
+            color: isDarkMode ? Colors.light : Colors.dark,
+          },
+        ]}>
+        {children}
+      </Text>
+    </View>
+  );
+};
 
-           <Text style={styles.inputTitle}>Senha</Text>
-           <TextInput style={styles.inputContainer}></TextInput>
-           
-           <Button title='Entrar' color="#841584"/>
-         </View>
-       </ScrollView>
-     </SafeAreaView>
-   );
- };
- 
- const styles = StyleSheet.create({
-   sectionContainer: {
-     marginTop: 32,
-     paddingHorizontal: 24,
-   },
-   sectionTitle: {
-     fontSize: 24,
-     fontWeight: 'bold',
-   },
-   sectionDescription: {
-     marginTop: 8,
-     fontSize: 18,
-     fontWeight: '400',
-   },
-   highlight: {
-     fontWeight: '700',
-   },
-   inputTitle: {
-     fontWeight: "400",
-     marginLeft: 15,
-   },
-   inputContainer: {
-     height: 60,
-     margin: 15,
-     marginBottom: 50,
-     padding: 10,
-     borderWidth: 2,
-     borderColor: "#AAAAAA",
-     borderRadius: 10,
-   },
- });
- 
- 
- export default App;
+const App = () => {
+  const isDarkMode = useColorScheme() === 'dark';
+
+  const backgroundStyle = {
+    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
+  };
+
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState(false);
+  const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState(false);
+
+  const emailValidator = new RegExp('[a-zA-Z0-9.]+@[a-zA-Z0-9]+[.][a-zA-Z]+([.][a-zA-Z]+)?');
+  const passwordValidator = new RegExp('(?=.*[0-9])(?=.*[a-zA-Z]).{7,}');
+
+  const validateInput = () => {
+    if(!emailValidator.test(email)){
+      setEmailError(true);
+    }
+    else{
+      setEmailError(false);
+    }
```

Comment:
```suggestion
  setEmailError(!emailValidator.test(email));
```
mesma coisa pro `passwordValidator`